File: frontend/src/types/provider.ts
export interface ProviderStatus {
  name: string;
  has_credentials: boolean;
  is_enabled: boolean;
  account_count: number;
  last_sync_time: string | null;
  supports_setup: boolean;
}

export interface ProviderSetupField {
  key: string;
  label: string;
  help_text: string;
  input_type: "text" | "textarea" | "password" | "select";
  options?: { value: string; label: string }[];
}

export interface ProviderSetupResponse {
  provider: string;
  message: string;
  warnings: string[];
}

File: backend/schemas/provider.py
"""Pydantic schemas for provider settings."""

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel


class ProviderStatusResponse(BaseModel):
    """Response schema for a single provider's status."""

    name: str
    has_credentials: bool
    is_enabled: bool
    account_count: int
    last_sync_time: Optional[datetime] = None

    model_config = {"from_attributes": True}


class ProviderUpdateRequest(BaseModel):
    """Request body for updating provider settings."""

    is_enabled: bool


class ProviderSetupRequest(BaseModel):
    """Request body for provider setup. Credentials vary by provider."""

    credentials: dict[str, str]


class ProviderSetupResponse(BaseModel):
    """Response after successful provider setup."""

    provider: str
    message: str


class ProviderCredentialInfo(BaseModel):
    """Describes a credential field for a provider's setup form."""

    key: str
    label: str
    help_text: str = ""
    input_type: Literal["text", "textarea", "password"] = "text"
